department-of-veterans-affairs/caseflow
# Sort by issue count

Resolves #11314

### Description
Allows the back end to sort by the number of issues on an appeal. Also adds `issue_count` to the cached appeal attributes table.

|Before|ASC|DESC|
|---|---|---|
|<img width="517" alt="Screen Shot 2019-08-07 at 4 52 58 PM" src="https://user-images.githubusercontent.com/45575454/62702084-daa45c80-b9b3-11e9-8ff2-e651a2306ccb.png">|<img width="500" alt="Screen Shot 2019-08-08 at 8 09 18 AM" src="https://user-images.githubusercontent.com/45575454/62702089-ded07a00-b9b3-11e9-833e-2b323e6abefe.png">|<img width="504" alt="Screen Shot 2019-08-08 at 8 09 12 AM" src="https://user-images.githubusercontent.com/45575454/62702097-e1cb6a80-b9b3-11e9-8be4-c270e2439c70.png">|




### Acceptance Criteria
 - [ ] Back end correctly stores each appeal's issue_count
 - [ ] Front end correctly shows the back end sorted tasks

### Testing Plan
1. Run the new migration with
```bash
bin/rails db:migrate
```
2. Run the updated job in the rails console with
```ruby
> UpdateCachedAppealsAttributesJob.perform_now
> CachedAppeal.find_by(appeal_id: 11, appeal_type: Appeal.name)
=> #<CachedAppeal:0x00007f864e2491a0
 appeal_id: 11,
 appeal_type: "Appeal",
 closest_regional_office_city: "St. Petersburg",
 docket_number: "190805-11",
 docket_type: "direct_review",
 vacols_id: nil,
 veteran_name: "Goodwin, Leo",
 issue_count: 1>
```
3. Add `Constants.QUEUE_CONFIG.ISSUE_COUNT_COLUMN` to `columns` in on_hold_tasks_tab.rb.
4. Change user to BVALSPORER and go to the /organizations/vlj-support "On hold" tab
5. Toggle the Issues sorting button

*Schema Changes Only*
